ext/RMagick/extconf.rb

```diff
@@ -123,31 +123,73 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def detect_imagemagick_packages(packages)
+      existed_packages = packages.map do |package|
+        if system "pkg-config --exists #{package}"
+          package
+        end
+      end
+      existed_packages.compact
```

Comment:
I think something like this might work:

```suggestion
      packages.select do |package|
        system "pkg-config --exists #{package}"
      end
```